Remove WebSQLiteDatabaseTracker in web process

szewai opened the pull request:
#### 4596c1459a9de783b6d52662c529912683c4828d
<pre>
Remove WebSQLiteDatabaseTracker in web process
<a href="https://bugs.webkit.org/show_bug.cgi?id=240598">https://bugs.webkit.org/show_bug.cgi?id=240598</a>

Reviewed by NOBODY (OOPS!).

Currently when web process starts performing SQLite database operations, it sends a message to UI process to ask UI
process to hold an assertion for it. This is unreliable because UI process might drop assertion and make web process
suspended before it receives the messasge. To avoid this, we apply the solution we made for network process to web
process. That is, aborting and suspending database operations when process receives PrepareToSuspend, and resuming
database operations after process receives ProcessDidResume. This patch also simplifies process suspesion logic since
UI process does not need to take assertion for web process for holding locked files.

* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::shutDown):
(WebKit::WebProcessProxy::setIsHoldingLockedFiles): Deleted.
* Source/WebKit/UIProcess/WebProcessProxy.h:
* Source/WebKit/UIProcess/WebProcessProxy.messages.in:
* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::m_nonVisibleProcessMemoryCleanupTimer):
(WebKit::WebProcess::prepareToSuspend):
(WebKit::WebProcess::processDidResume):
(WebKit::m_webSQLiteDatabaseTracker): Deleted.
* Source/WebKit/WebProcess/WebProcess.h:
No new tests (OOPS!).
* Source/WebCore/loader/appcache/ApplicationCacheStorage.cpp:
(WebCore::ApplicationCacheStorage::openDatabase):
(WebCore::ApplicationCacheStorage::setAllowsOpeningDatabase):
* Source/WebCore/loader/appcache/ApplicationCacheStorage.h:
* Source/WebCore/platform/sql/SQLiteDatabase.cpp:
(WebCore::SQLiteDatabase::open):
(WebCore::WTF_GUARDED_BY_LOCK): Deleted.
(WebCore::SQLiteDatabase::setIsDatabaseOpeningForbidden): Deleted.
* Source/WebCore/platform/sql/SQLiteDatabase.h:
</pre>

rniwa commented on Source/WebCore/Modules/webdatabase/DatabaseTracker.cpp:
```diff
@@ -1337,5 +1337,4 @@ void DatabaseTracker::notifyDatabasesChanged()
         tracker.m_client->dispatchDidModifyDatabase(notification.first, notification.second);
 }
 
-
```
Seems unnecessary now?

szewai committed:
Remove WebSQLiteDatabaseTracker in web process
https://bugs.webkit.org/show_bug.cgi?id=240598

Reviewed by NOBODY (OOPS!).

Currently when web process starts performing SQLite database operations, it sends a message to UI process to ask UI
process to hold an assertion for it. This is unreliable because UI process might drop assertion and make web process
suspended before it receives the messasge. To avoid this, we apply the solution we made for network process to web
process. That is, aborting and suspending database operations when process receives PrepareToSuspend, and resuming
database operations after process receives ProcessDidResume. This patch also simplifies process suspesion logic since
UI process does not need to take assertion for web process for holding locked files.

* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::shutDown):
(WebKit::WebProcessProxy::setIsHoldingLockedFiles): Deleted.
* Source/WebKit/UIProcess/WebProcessProxy.h:
* Source/WebKit/UIProcess/WebProcessProxy.messages.in:
* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::m_nonVisibleProcessMemoryCleanupTimer):
(WebKit::WebProcess::prepareToSuspend):
(WebKit::WebProcess::processDidResume):
(WebKit::m_webSQLiteDatabaseTracker): Deleted.
* Source/WebKit/WebProcess/WebProcess.h:
No new tests (OOPS!).
* Source/WebCore/loader/appcache/ApplicationCacheStorage.cpp:
(WebCore::ApplicationCacheStorage::openDatabase):
(WebCore::ApplicationCacheStorage::setAllowsOpeningDatabase):
* Source/WebCore/loader/appcache/ApplicationCacheStorage.h:
* Source/WebCore/platform/sql/SQLiteDatabase.cpp:
(WebCore::SQLiteDatabase::open):
(WebCore::WTF_GUARDED_BY_LOCK): Deleted.
(WebCore::SQLiteDatabase::setIsDatabaseOpeningForbidden): Deleted.
* Source/WebCore/platform/sql/SQLiteDatabase.h:
